### PR TITLE
Simplified & Unified conditional configuration.

### PR DESCRIPTION
**Pull request by seancpeters**

### CODE REVIEW

File: src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/ConditionalBlockCommentConfig.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    public static class ConditionalBlockCommentConfig
+    {
+        public static List<IOperationProvider> ConfigureFromJObject(JObject rawConfiguration)
+        {
+            string startComment = rawConfiguration.ToString("startComment");
+            string endComment = rawConfiguration.ToString("endComment");
+
+            if (string.IsNullOrWhiteSpace(startComment) || string.IsNullOrWhiteSpace(endComment))
+            {
+                throw new Exception($"Template authoring error. StartComment and EndComment must be defined");
```
Comment (mlorbetske) [inline src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/ConditionalBlockCommentConfig.cs]: With the hardening effort - we should really make these custom exception types & catch them at the outer run to make sure no traces are displayed but the right message is conveyed to the author (as these will likely get picked off in testing)

File: src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/ConditionalBlockCommentConfig.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+using Microsoft.TemplateEngine.Core.Operations;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Config
+{
+    public static class ConditionalBlockCommentConfig
+    {
+        public static List<IOperationProvider> ConfigureFromJObject(JObject rawConfiguration)
+        {
+            string startComment = rawConfiguration.ToString("startComment");
```
Comment (mlorbetske) [inline src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/ConditionalBlockCommentConfig.cs]: I think "comment" is wrong for these in general, as we support reading effectively "running" preprocessors - maybe this should be "startSyntax"/"endSyntax" or something along those lines (similar for single line setups)

Comment (seancpeters) [inline src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/ConditionalBlockCommentConfig.cs]: Maybe "startToken" / "endToken" ?

File: src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/ConditionalBlockCommentConfig.cs (hunk 1 above)
Comment (seancpeters) [inline src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/ConditionalBlockCommentConfig.cs]: 👍

File: src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/ConditionalBlockCommentConfig.cs (hunk 2 above)
Comment (mlorbetske) [inline src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/ConditionalBlockCommentConfig.cs]: that'll work

**Pull request merged**